src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs

```diff
@@ -607,7 +606,10 @@ internal static OperationStatus EncodeUtf8Shim(TextEncoder encoder, ReadOnlySpan
                         return OperationStatus.DestinationTooSmall;
                     }
 
-                    return EncodeIntoBuffer(destinationPtr, destination.Length, sourcePtr, source.Length, out charsConsumed, out charsWritten, firstCharacterToEncode, isFinalBlock);
+                    fixed (char* destinationPtr = destination)
```

Comment:
What would be the benefit of deferring creating this fixed block? In general, is it best to keep your fixed blocks as small as possible and is there no value in trying to batch them together with other fixed block setup?

cc @jkotas